spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/data/redis/RedisProperties.java

```diff
@@ -76,6 +76,11 @@
 	 */
 	private String clientName;
 
+	/**
+	 * Client type to use - 'jedis' or 'lettuce'.
+	 */
+	private String clientType;
```

Comment:
I'd use an enum with `Jedis` and `Lettuce` here rather than a raw string. This also improves the developer experience in the IDE.

Reply:
I will wait to see what shakes out of https://github.com/spring-projects/spring-boot/issues/22559#issuecomment-664897053 before updating.

Reply:
Thanks @bono007. We've decide to go ahead with the property so if you have time to update your proposal to use an enum that would be much appreciated.

Reply:
Sounds good @snicoll. The property approach will be cleaner than the "move the auto-configs to the connection configs" I was proposing as that ended up being a bit messier than I originally thought when I prototyped it. 

I will get to it this evening.

Reply:
Sounds good @snicoll. The property approach will be cleaner than the "move the auto-configs to the connection configs" I was proposing as that ended up being a bit messier than I originally thought when I prototyped it.

I will get to it this evening.

Reply:
Done. I squashed and rebased as well.